cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AppFabricServiceRuntimeModule.java

```diff
@@ -344,8 +344,8 @@ protected void configure() {
     @Named(Constants.AppFabric.SERVER_ADDRESS)
     @SuppressWarnings("unused")
     public InetAddress providesHostname(CConfiguration cConf) {
-      return Networks.resolve(cConf.get(Constants.AppFabric.SERVER_ADDRESS),
-                              new InetSocketAddress("localhost", 0).getAddress());
+      String address = cConf.get(Constants.AppFabric.SERVER_ADDRESS);
```

Comment:
Agreed, this is not changing functionality, simply to avoid inlining the usage of the address. I can revert if the preference is to keep the resolution inline.